catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -293,6 +308,68 @@ public void testRouteCreationMissingMonitoredDirectory() throws Exception {
         empty());
   }
 
+  @Test
+  public void testInterruptedWhileWaitingForInputTransformers() {
+    Thread.currentThread().interrupt();
+
+    ContentDirectoryMonitor monitor =
+        new ContentDirectoryMonitor(
+            camelContext, mock(AttributeRegistry.class), 1, 1, Runnable::run);
+    monitor.systemSubjectBinder = exchange -> {};
+    monitor.setNumThreads(1);
+    monitor.setReadLockIntervalMilliseconds(1000);
+    monitor.setMonitoredDirectoryPath(monitoredDirectoryPath);
+
+    monitor.setInputTransformers(Collections.emptyList());
+
+    monitor.attemptAddRoutes();
+    assertThat(
+        "The content directory monitor should have no route definitions",
+        monitor.getRouteDefinitions(),
+        hasSize(0));
+  }
+
+  @Test
+  public void testInitWithConfigDeleted() {
+    ContentDirectoryMonitor monitor =
+        new ContentDirectoryMonitor(
+            camelContext, mock(AttributeRegistry.class), 1, 1, Runnable::run);
+    monitor.systemSubjectBinder = exchange -> {};
+    monitor.setNumThreads(1);
+    monitor.setReadLockIntervalMilliseconds(1000);
+    monitor.setMonitoredDirectoryPath(monitoredDirectoryPath);
+
+    monitor.setConfigDeleted(true);
+    monitor.attemptAddRoutes();
```

Comment:
❓`attemptAddRoutes()` should be private in `ContentDirectoryMontior`, right? Is there a way to test that method by calling a public method (i.e. `init()`) instead?